test/interpreter/function-address.spec.ts

```diff
@@ -0,0 +1,72 @@
+import {HyperFormula, ErrorType} from '../../src'
+import {adr, detailedError} from '../testUtils'
+import {ErrorMessage} from '../../src/error-message'
+
+describe('ADDRESS', () => {
+  it('with row and col', () => {
+    const engine = HyperFormula.buildFromArray([
+      ['=ADDRESS(1,1)'],
+      ['=ADDRESS(77,300)'],
+      ['=ADDRESS(ROW(),300)'],
+      ['=ADDRESS(45,COLUMN())'],
+    ])
+
+    expect(engine.getCellValue(adr('A1'))).toEqual('$A$1')
+    expect(engine.getCellValue(adr('A2'))).toEqual('$KN$77')
+    expect(engine.getCellValue(adr('A3'))).toEqual('$KN$3')
+    expect(engine.getCellValue(adr('A4'))).toEqual('$A$45')
+  })
+
+  it('with row, col, and abs', () => {
+    const engine = HyperFormula.buildFromArray([
+      ['=ADDRESS(1,1,1)'],
+      ['=ADDRESS(1,1,2)'],
+      ['=ADDRESS(1,1,3)'],
+      ['=ADDRESS(1,1,4)'],
+    ])
+
+    expect(engine.getCellValue(adr('A1'))).toEqual('$A$1')
+    expect(engine.getCellValue(adr('A2'))).toEqual('A$1')
+    expect(engine.getCellValue(adr('A3'))).toEqual('$A1')
+    expect(engine.getCellValue(adr('A4'))).toEqual('A1')
+  })
+
+  it('with row, col, abs, and sheetName', () => {
+    const engine = HyperFormula.buildFromArray([
+      ['=ADDRESS(1,1,1, TRUE(), "Sheet1")'],
+      ['=ADDRESS(1,1,2, TRUE(), "Sheet2")'],
+      ['=ADDRESS(1,1,3, TRUE(), "Sheet3")'],
+      ['=ADDRESS(1,1,4, TRUE(), "Sheet4")'],
+      ['=ADDRESS(1,1,4, TRUE(), "")'],
+    ])
+
+    expect(engine.getCellValue(adr('A1'))).toEqual('Sheet1!$A$1')
+    expect(engine.getCellValue(adr('A2'))).toEqual('Sheet2!A$1')
+    expect(engine.getCellValue(adr('A3'))).toEqual('Sheet3!$A1')
+    expect(engine.getCellValue(adr('A4'))).toEqual('Sheet4!A1')
+    expect(engine.getCellValue(adr('A5'))).toEqual('!A1')
+  })
+
+
+  it('invalid arguments', () => {
+    const engine = HyperFormula.buildFromArray([
+      ['=ADDRESS()'],
+      ['=ADDRESS(1)'],
+      ['=ADDRESS(0,0)'],
+      ['=ADDRESS("row1","row2")'],
+      ['=ADDRESS(1,1,0)'],
+      ['=ADDRESS(1,1,5)'],
+      ['=ADDRESS(1,1,4,FALSE())'],
+      ['=ADDRESS(1,1,1, true, "")'],
+    ])
+
+    expect(engine.getCellValue(adr('A1'))).toEqualError(detailedError(ErrorType.NA, ErrorMessage.WrongArgNumber))
+    expect(engine.getCellValue(adr('A2'))).toEqualError(detailedError(ErrorType.NA, ErrorMessage.WrongArgNumber))
+    expect(engine.getCellValue(adr('A3'))).toEqualError(detailedError(ErrorType.NUM, ErrorMessage.ValueSmall))
+    expect(engine.getCellValue(adr('A4'))).toEqualError(detailedError(ErrorType.VALUE, ErrorMessage.NumberCoercion))
+    expect(engine.getCellValue(adr('A5'))).toEqualError(detailedError(ErrorType.NUM, ErrorMessage.ValueSmall))
+    expect(engine.getCellValue(adr('A6'))).toEqualError(detailedError(ErrorType.NUM, ErrorMessage.ValueLarge))
+    expect(engine.getCellValue(adr('A7'))).toEqualError(detailedError(ErrorType.NA, ErrorMessage.ArgumentMustEqual('useA1Style', 'TRUE')))
```

Comment:
In my opinion, we should implement the conversion to R1C1 as well. It doesn't require a lot of effort but will improve the compatibility with both Excel and Sheets.

Reply:
Added support for R1C1 notation - Passing TRUE() for the 4th argument will no longer result in an error. Updated unit tests; Verified coverage; Updated the information in https://github.com/handsontable/hyperformula/issues/1221.